pkg/security/mcs/label_test.go

```diff
@@ -0,0 +1,141 @@
+package mcs
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestParseRange(t *testing.T) {
+	testCases := map[string]struct {
+		in    string
+		errFn func(error) bool
+		r     Range
+		total uint64
+	}{
+		"identity range": {
+			in: "test,s0/1",
+			r: Range{
+				prefix: "test,s0",
+				k:      1,
+			},
+			total: 1024,
+		},
+		"simple range": {
+			in: "s0/2",
+			r: Range{
+				prefix: "s0",
+				k:      2,
+			},
+			total: 523776,
+		},
+		"NaN": {
+			in:    "/a",
+			errFn: func(err error) bool { return strings.Contains(err.Error(), "range not in the format") },
+		},
+		"zero k": {
+			in:    "/0",
+			errFn: func(err error) bool { return strings.Contains(err.Error(), "label length must be a positive integer") },
+		},
+	}
+
+	for s, testCase := range testCases {
+		r, err := ParseRange(testCase.in)
+		if testCase.errFn != nil && !testCase.errFn(err) {
+			t.Errorf("%s: unexpected error: %v", s, err)
+			continue
+		}
+		if err != nil {
+			continue
+		}
+		if r.prefix != testCase.r.prefix || r.k != testCase.r.k {
+			t.Errorf("%s: unexpected range: %#v", s, r)
+		}
+		if r.Size() != testCase.total {
+			t.Errorf("%s: unexpected total: %d", s, r.Size())
+		}
+	}
+}
+
+func TestLabel(t *testing.T) {
+	testCases := map[string]struct {
+		in     string
+		expect Categories
+		offset uint64
+		out    string
+	}{
+		"identity range": {
+			in:     "c0,c1",
+			expect: Categories{1, 0},
+			offset: 0,
+			out:    "c1,c0",
```

Comment:
I don't know what your reason is for high->low ordering on out. The kernel will transform these into low->high.  It will also translate contiguous categories into `.` notation. But you do not need to do so here for any reason.